File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashMapS = {}
        hashMapT = {}
        
        for c in s:
            if c in hashMapS:
                hashMapS[c] += 1
            else:
                hashMapS[c] = 1
        
        for c in t:
            if c in hashMapT:
                hashMapT[c] += 1
            else:
                hashMapT[c] = 1
        
        return hashMapS == hashMapT

File: 0242-valid-anagram/0242-valid-anagram.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> hashMapS = new HashMap();
        Map<Character, Integer> hashMapT = new HashMap();
        
        for (char c:s.toCharArray()){
            hashMapS.put(c, hashMapS.getOrDefault(c, 0)+1);
        }
        for (char c:t.toCharArray()){
            hashMapT.put(c, hashMapT.getOrDefault(c, 0)+1);
        }
        return hashMapS.equals(hashMapT);
    }
}